accord-core/src/main/java/accord/topology/TopologyManager.java

```diff
@@ -371,17 +395,86 @@ private int indexOf(long epoch)
         }
     }
 
+    private static class FutureEpoch
+    {
+        private volatile AsyncResult.Settable<Void> future;
+        private long deadlineMillis;
+
+        public FutureEpoch(long deadlineMillis)
+        {
+            this.future = AsyncResults.settable();
+            this.deadlineMillis = deadlineMillis;
```

Comment:
shouldn't we use `nano` rather than `millis`?  `millis` can go backwards but `nano` can't.

Reply:
That makes sense. We don't expose that to Accord right now, but I could add it as something like `elapsedTime` which is elapsed time since some arbitrary point and then use that for things that rely on time. For test code it would be the same time progression as `currentTimeMillis`

Durability scheduling also uses `currentTimeMillis` indirectly so it would also benefit.

Reply:
> Durability scheduling 

as far as I can tell that is the only caller of `unixTime(TimeUnit)` and it doesn't need `currentTimeMillis` and is better as `nanoTime` (as you are pointing out)... so moving `unixTime(TimeUnit)` to `elapsedTime` makes sense to me...